avocado/utils/configure_network.py

```diff
@@ -35,111 +35,176 @@ class NWException(Exception):
             raise NWException("ifup fails: %s" % ex)
 
+    def interface_down(self):
+        cmd = "ifdown %s" % self._interface
+        try:
+            process.system(cmd, sudo=True)
+        except Exception as ex:
+            raise NWException("ifdown fails: %s" % ex)
 
-def unset_ip(interface):
-    """
-    Gets interface name unassigns the IP to the interface
-    """
-    if distro.detect().name == 'rhel':
-        conf_file = "/etc/sysconfig/network-scripts/ifcfg-%s" % interface
-
-    if distro.detect().name == 'SuSE':
-        conf_file = "/etc/sysconfig/network/ifcfg-%s" % interface
+    def _move_interfaceconfile(self, src_conf, dest_conf):
+        if os.path.exists(src_conf):
+            shutil.move(src_conf, dest_conf)
+        else:
+            raise NWException("%s interface not available" % self._interface)
 
-    cmd = "ifdown %s" % interface
-    try:
-        process.system(cmd, sudo=True)
-        shutil.move(conf_file+".backup", conf_file)
-        return True
-    except Exception as ex:
-        raise NWException("ifdown fails: %s" % ex)
+    def get_hwaddr(self):
+        f = open('/sys/class/net/%s/address' % self._interface)
+        hwaddr = f.read().strip()
+        f.close()
+        return hwaddr
 
+    def set_hwaddr(self, hwaddr):
+        """
+        :param hwaddr: Pass Hardwae address for defined interface
+        """
+        try:
+            process.system('ifconfig %s hw ether %s' %
+                           (self._interface, hwaddr))
+        except Exception as ex:
+            raise NWException("ifdown fails: %s" % ex)
 
-def ping_check(interface, peer_ip, count, option=None, flood=False):
-    """
-    Checks if the ping to peer works.
-    """
-    cmd = "ping -I %s %s -c %s" % (interface, peer_ip, count)
-    if flood is True:
-        cmd = "%s -f" % cmd
-    elif option is not None:
-        cmd = "%s %s" % (cmd, option)
-    if process.system(cmd, shell=True, verbose=True,
-                      ignore_status=True) != 0:
+    def add_maddr(self, maddr):
+        """
+        :param maddr: Mac address
+        :return: True /False Based on success
+        """
+        try:
+            process.system('ip maddr add %s dev %s' % (maddr, self._interface))
+            return True
+        except Exception as ex:
+            raise NWException("ifdown fails: %s" % ex)
         return False
-    return True
-
 
-def set_mtu_host(interface, mtu):
-    """
-    Set MTU size in host interface
-    """
-    cmd = "ip link set %s mtu %s" % (interface, mtu)
-    try:
-        process.system(cmd, shell=True)
-    except process.CmdError as ex:
-        raise NWException("MTU size can not be set: %s" % ex)
-    try:
-        cmd = "ip add show %s" % interface
-        mtuvalue = process.system_output(cmd, shell=True).decode("utf-8") \
-                                                         .split()[4]
-        if mtuvalue == mtu:
+    def del_maddr(self, maddr):
```

Comment:
suggestion: `remove_hardware_address()` or `remove_mac_address()`

Also, let's try to not abbreviate the variables/attributes. It makes the code clean. Explicit is better than implicit.